components/raftstore/src/store/worker/pd.rs

```diff
@@ -2497,51 +2468,16 @@ fn collect_report_read_peer_stats(
     stats
 }
 
-fn collect_engine_size<EK: KvEngine, ER: RaftEngine>(
-    coprocessor_host: &CoprocessorHost<EK>,
-    store_info: Option<&StoreInfo<EK, ER>>,
-    snap_mgr_size: u64,
-) -> Option<(u64, u64, u64)> {
+fn collect_engine_size<EK: KvEngine>(coprocessor_host: &CoprocessorHost<EK>) -> (u64, u64, u64) {
     if let Some(engine_size) = coprocessor_host.on_compute_engine_size() {
-        return Some((engine_size.capacity, engine_size.used, engine_size.avail));
-    }
-    let store_info = store_info.unwrap();
-    let (disk_cap, disk_avail) = match get_disk_space_stats(store_info.kv_engine.path()) {
-        Err(e) => {
-            error!(
-                "get disk stat for rocksdb failed";
-                "engine_path" => store_info.kv_engine.path(),
-                "err" => ?e
-            );
-            return None;
-        }
-        Ok((total_size, available_size)) => (total_size, available_size),
-    };
-    let capacity = if store_info.capacity == 0 || disk_cap < store_info.capacity {
-        disk_cap
+        (engine_size.capacity, engine_size.used, engine_size.avail)
     } else {
-        store_info.capacity
-    };
-    let raft_size = store_info
-        .raft_engine
-        .get_engine_size()
-        .expect("raft engine used size");
-
-    let kv_size = store_info
-        .kv_engine
-        .get_engine_used_size()
-        .expect("kv engine used size");
-
-    STORE_SIZE_EVENT_INT_VEC.raft_size.set(raft_size as i64);
-    STORE_SIZE_EVENT_INT_VEC.snap_size.set(snap_mgr_size as i64);
-    STORE_SIZE_EVENT_INT_VEC.kv_size.set(kv_size as i64);
-
-    let used_size = snap_mgr_size + kv_size + raft_size;
```

Comment:
Previously, `collect_engine_size` didn't account for `placeholder_size` and `separated_raft_mount_path` in the `used_size` calculation. This PR fixes that by consolidating similar logic into `DiskUsageChecker::inspect`, which is a good improvement.